library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -83,15 +96,31 @@ internal class TransactionPayloadFragment :
     override fun onDestroyView() {
         super.onDestroyView()
         uiLoaderTask?.cancel(true)
+        fileSaverTask?.cancel(true)
     }
 
     override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
-        if ((type == TYPE_RESPONSE || type == TYPE_REQUEST) && body.text.isNotEmpty()) {
-            val searchMenuItem = menu.findItem(R.id.search)
-            searchMenuItem.isVisible = true
-            val searchView = searchMenuItem.actionView as SearchView
-            searchView.setOnQueryTextListener(this)
-            searchView.setIconifiedByDefault(true)
+        if ((type == TYPE_RESPONSE || type == TYPE_REQUEST)) {
+            if (body.text.isNotEmpty()) {
+                val searchMenuItem = menu.findItem(R.id.search)
+                searchMenuItem.isVisible = true
+                val searchView = searchMenuItem.actionView as SearchView
+                searchView.setOnQueryTextListener(this)
+                searchView.setIconifiedByDefault(true)
+            }
+
+            val saveMenuItem = menu.findItem(R.id.save_body)
```

Comment:
What if we follow the same principle of not showing an icon in case of empty body, like it is done with search feature? If we do so, we can also remove complex conditions from `viewBodyExternally`, because there will be no need for toasts with errors.

Reply:
Agree, I've added it. Dont' you mind taking a look?

Reply:
Sure